domain/application/state/state.go

```diff
@@ -28,7 +29,33 @@ type commonStateBase struct {
 	*domain.StateBase
 }
 
-func (s *commonStateBase) checkChamReferenceExists(ctx context.Context, tx *sqlair.TX, referenceName string, revision int) (corecharm.ID, error) {
+func (s *commonStateBase) checkCharmExists(ctx context.Context, tx *sqlair.TX, id charmID) error {
```

Comment:
Since this is being added to all those functions, it would be good to test that they also give this error (if this is not tested already).

Reply:
I'll double-check :+1: